### PR TITLE
Make expected input image configurable

### PR DESCRIPTION
**Pull request by aftersomemath**
Unfortunately the sim can only publish RGBA images which makes this necessary.

### CODE REVIEW

**Comment by aftersomemath** [comment]
Tested on drone.

File: src/GridLineEstimator.cpp [hunk 1]
```diff
@@ -118,6 +119,19 @@ GridLineEstimator::GridLineEstimator(
...
                             line_extractor_settings_.hough_theta_resolution,
                             0.0);
+
+    if (expected_image_format == "RGB") {
+        hsv_conversion_constant_ = CV_RGB2HSV;
+        image_encoding_ = sensor_msgs::image_encodings::RGB8;
+    }
+    else if (expected_image_format == "RGBA") {
+        // HSV does not have a CV_RGBA2HSV constant defined
+        hsv_conversion_constant_ = CV_RGB2HSV;
```
Comment (amiller27) [inline src/GridLineEstimator.cpp:129]: Does this work?

Comment (amiller27) [reply src/GridLineEstimator.cpp:129]: Also why do we need this variable if it's the same in all cases?

Comment (aftersomemath) [reply src/GridLineEstimator.cpp:129]: We don't but I did it for consistency. Also yes it works.

**Review by amiller27 — APPROVED** [review]

**Pull request merged**